implementations/rust/ockam/ockam_abac/src/policy.rs

```diff
@@ -82,10 +126,134 @@ impl IncomingAccessControl for PolicyAccessControl {
         AbacAccessControl::new(
             self.identities_attributes.clone(),
             self.authority.clone(),
-            policy,
+            policy_expression,
             self.environment.clone(),
         )
         .is_authorized(msg)
         .await
     }
 }
+
+pub struct Policies {
+    resources_repository: Arc<dyn ResourcePoliciesRepository>,
+    resource_types_repository: Arc<dyn ResourceTypePoliciesRepository>,
+}
+
+impl Policies {
+    pub fn new(
+        resources_repository: Arc<dyn ResourcePoliciesRepository>,
+        resource_types_repository: Arc<dyn ResourceTypePoliciesRepository>,
+    ) -> Self {
+        Self {
+            resources_repository,
+            resource_types_repository,
+        }
+    }
+
+    async fn get_expression_for_resource_type_and_name(
+        &self,
+        resource: &Resource,
+        action: &Action,
+    ) -> Result<Expr> {
+        let mut expression = if let Some(policy) = self
+            .resources_repository
+            .get_policy(&resource.resource_name, action)
+            .await?
+        {
+            policy.into_expression()
+        } else {
+            return Err(Error::new(
+                Origin::Application,
+                Kind::Misuse,
+                format!(
+                    "No policy found for resource: {} and action: {}",
+                    resource.resource_name, action
+                ),
+            ));
+        };
+        if let Some(policy) = self
+            .resource_types_repository
+            .get_policy(&resource.resource_type, action)
+            .await?
+        {
+            expression = and([policy.into_expression(), expression]);
```

Comment:
I'm not saying it doesn't make sense but for some reason I was expecting resource policy to just overwrite the resource type policy (if any). Let's maybe think which behavior makes sense more

Reply:
```
 I can imagine a use case where we want to have a common policy for all the outlets, but then have a specific rule that applies to each of them uniquely.
```
Sure. Let's say you have an outlet outlet1 with policy policy1, also all outlets have policy policy2. So what should be the policy for outlet1 in the end? policy1 or policy1&policy2? I would expect the first one, but I see that your code doing the latter. So which one we should pick?